cmd/gapit/validate_gpu_profiling.go

```diff
@@ -51,11 +51,15 @@ func (verb *validateGpuProfilingVerb) Run(ctx context.Context, flags flag.FlagSe
 	someDeviceFailed := false
 	for i, p := range devices {
 		fmt.Fprintf(stdout, "-- Device %v: %v --\n", i, p.ID.ID())
-		err = client.ValidateDevice(ctx, p)
+		res, err := client.ValidateDevice(ctx, p)
 		if err != nil {
 			fmt.Fprintf(stdout, "%v\n", log.Err(ctx, err, "Failed to validate device"))
 			someDeviceFailed = true
 			continue
+		} else if len(res.ValidationErrorMsg) > 0 {
+			fmt.Fprintf(stdout, "%v\n", log.Errf(ctx, nil, "Failed to validate device: %s", res.ValidationErrorMsg))
```

Comment:
We should show different messages here for the two error cases. Maybe `Failed to validate device: <error>` and `Device did not pass validation: <msg>, trace file: <trace file>`.

Also, no need to turn the string into an error to then turn it back into a string.